lib/katello/tasks/update_subscription_facet_backend_data.rake

```diff
@@ -0,0 +1,61 @@
+namespace :katello do
+  task :update_subscription_facet_backend_data => ["environment"] do
+    def error(message)
+      @errors << message
+      @errors << "\n"
+    end
+
+    def report_errors
+      if @errors.any?
+        filename = "subscription_facet_upgrade-#{Time.now.to_i}.log"
+        path = "/var/log/foreman/#{filename}"
```

Comment:
I'd note that this is simply a move of an existing file to a new location so it could be called in a shared manner. I'm hesitant to re-factor this code here since it is/has been used with the 3.0 upgrades and will be used again. A stand-alone update to this upgrade task to re-factor it might make more sense.